### PR TITLE
remove individual sitrep font size. use general font size instead

### PR DESCRIPTION
**Pull request by TheSilentOne1**
in reference to:
https://github.com/freeorion/freeorion/issues/901

### CODE REVIEW

**Comment by adrianbroher** [comment]
Just a note that is unrelated to the PR content: did you intent to use this author name? 
Because it isn't associated with your github account.  It doesn't really matter from a technical standpoint, but you either have used another author name for earlier commits and may want to configure your local git installation accordingly or or you want associate the new email with github.  It will still work if you choose to do nothing.

**Comment by TheSilentOne1** [comment]
Thanks for pointing that out. Think I should've changed my local git client to the right account now.

File: UI/SitRepPanel.cpp [hunk 1]
```diff
@@ -269,7 +265,7 @@ namespace {
             AttachChild(m_icon);
 
             m_link_text = new LinkText(GG::X0, GG::Y0, GG::X1, m_sitrep_entry.GetText() + " ", 
-                                       ClientUI::GetFont(std::max(1.0*ClientUI::Pts(), 1.0*GetOptionsDB().Get<int>("UI.sitrep-font-size"))),
+                                       ClientUI::GetFont(1.0*ClientUI::Pts()),
```
Comment (geoffthemedio) [inline UI/SitRepPanel.cpp]: no need for 1.0\* or any parameter to GetFont... the no-parameter version defaults to the same thing.

File: UI/SitRepPanel.cpp [hunk 2]
```diff
@@ -25,16 +25,12 @@ namespace {
...
         db.Add("UI.sitrep-icon-size", UserStringNop("OPTIONS_DB_UI_SITREP_ICONSIZE"), 24, RangedValidator<int>(12, 64));
-        db.Add("UI.sitrep-font-size", UserStringNop("OPTIONS_DB_UI_SITREP_FONTSIZE"), 12, RangedValidator<int>(10, 48));
     }
     bool temp_bool = RegisterOptions(&AddOptions);
 
     GG::X GetIconSize()
     { return GG::X(GetOptionsDB().Get<int>("UI.sitrep-icon-size")); }
 
-    GG::X GetFontSize()
-    { return GG::X(GetOptionsDB().Get<int>("UI.sitrep-font-size")); }
```
Comment (geoffthemedio) [inline UI/SitRepPanel.cpp]: if removing the use of this option, remove the option addition as well? and any relevant stringtable entry...

File: UI/SitRepPanel.cpp [hunk 3]
```diff
@@ -25,16 +25,12 @@ namespace {
         db.Add("verbose-sitrep", UserStringNop("OPTIONS_DB_VERBOSE_SITREP_DESC"),  false,  Validator<bool>());
         db.Add<std::string>("hidden-sitrep-templates", UserStringNop("OPTIONS_DB_HIDDEN_SITREP_TEMPLATES_DESC"), "");
         db.Add("UI.sitrep-icon-size", UserStringNop("OPTIONS_DB_UI_SITREP_ICONSIZE"), 24, RangedValidator<int>(12, 64));
-        db.Add("UI.sitrep-font-size", UserStringNop("OPTIONS_DB_UI_SITREP_FONTSIZE"), 12, RangedValidator<int>(10, 48));
```
Comment (geoffthemedio) [inline UI/SitRepPanel.cpp:28]: is there a stringtable entry to remove as well?

**Comment by TheSilentOne1** [comment]
Good catch with the stringtable entry. Have made the relevant changes.

**Comment by Vezzra** [comment]
Looks ok to me.

File: UI/SitRepPanel.cpp (hunk 1 above)
Comment (geoffthemedio) [reply UI/SitRepPanel.cpp]: still unnecessarily calling multiple functions... just ClientUI::GetFont() should suffice

File: UI/SitRepPanel.cpp [hunk 4]
```diff
@@ -269,7 +265,7 @@ namespace {
             AttachChild(m_icon);
 
             m_link_text = new LinkText(GG::X0, GG::Y0, GG::X1, m_sitrep_entry.GetText() + " ", 
-                                       ClientUI::GetFont(std::max(1.0*ClientUI::Pts(), 1.0*GetOptionsDB().Get<int>("UI.sitrep-font-size"))),
+                                       ClientUI::GetFont(ClientUI::Pts()),
```
Comment (geoffthemedio) [inline UI/SitRepPanel.cpp]: not sure what changed, but it wasn't this being fixed

Comment (Vezzra) [reply UI/SitRepPanel.cpp]: The `1.0*` directly before `ClientUI::Pts()` was still there and has been removed. But I guess what you meant is that `ClientUI::Pts()` can be removed as well, as `ClientUI::GetFont()` without any parameters will yield the exact same result anyway, correct?

Comment (geoffthemedio) [reply UI/SitRepPanel.cpp]: yes

**Comment by TheSilentOne1** [comment]
Removed that part as well, sorry.

**Pull request merged**